Processing/VALL_Deep_Dive.Rmd

```diff
@@ -0,0 +1,404 @@
+
+# However there is a new well POD in NM db, this is likely the real site location:
+proposed_well <- getPODNewMexico(aoi = park_boundary, dist = 0.0) %>%
+  dplyr::filter(park_right == "Park") %>%
+  mutate(source_type = "Shallow Groundwater",
+         type = "Well",
+         source_longitude = geometry[[1]][1],
+         sorce_latitude = geometry[[1]][[2]]) 
+
+well_aoi <- getXYWatersheds(sf = proposed_well, snap = FALSE, clip = TRUE)
+```
+
+### 3. Additional data sets
+
+NPS buildings
+
+```{r}
+buildings <- getBuildings(park_boundary = park_boundary, dist = 0) %>%
+  st_centroid(.)
+```
+
+Weather stations
+
+```{r}
+startDate <- "first"
+endDate <- Sys.Date()
+variables <- "all"
+
+#filter to aoi
+stations <-rnoaa::ghcnd_stations() %>%
+  # Station data in WGS84
+  sf::st_as_sf(coords = c("longitude", "latitude"), crs = 4326) %>%
+  # transform to park boundary (NAD83)
+  sf::st_transform(st_crs(park_boundary)) %>%
+  sf::st_filter(park_boundary %>% st_buffer(0.01)) %>%
+  dplyr::group_by(id, elevation, name, geometry, first_year) %>%
+  dplyr::summarize() %>%
+  sf::st_join(park_boundary) %>%
+  distinct(.keep_all = TRUE)
+
+
+# if startDate = 'first', find min year in the station metadata
+if (startDate == 'first'){
+  startDate <- paste(min(stations$first_year), "01", "01", sep = "-")
+}
+
+station_aoi <- stations %>%
+  dplyr::group_by(id, elevation, name, geometry) %>%
+  dplyr::summarize() %>%
+  distinct(.keep_all = TRUE)
+
+#get weather data for all stations (depending on dates and how many stations this can take a while)
+weather_data <- rnoaa::meteo_pull_monitors(station_aoi$id, date_min = startDate,
+                                           date_max = endDate,
+                                           var = variables) %>%
+  dplyr::filter(id == "USW00003062") %>%
+  dplyr::select(c(id, date, prcp, snow, snwd, tmax, tmin, tobs)) %>%
+  # info stored as data * 10 for these params... so reconvert:
+  dplyr::mutate_at(c("tmax", "tmin", "prcp"), function(x) {x * 0.10}) %>%
+  # convert from mm to inches
+  dplyr::mutate_at(c("snow","snwd", "prcp"), function(x) {x * 0.0393701}) %>%
+  # convert from Celsius to Fahrenheit
+  dplyr::mutate_at(c("tmax","tmin"), function(x) {(9/5) * x + 32})
+```
+
+Nearest NWIS Sites
+
+For many parks, the nearest USGS stream gage is far away. Therefore, here we pull in all NWIS gages within 100 km of the park boundary. Of those, we only select stream gages that are considered "reference" gages in the [GAGES-II database (Falcone, 2011)](https://pubs.usgs.gov/publication/70046617). We then select the gage that is closest to the park with the most complete period of record and delineate its associated watershed using the `get_nldi_basin()` function from the {nhdplusTools} package.
+
+```{r, eval = TRUE, message = FALSE, warning = FALSE}
+# Get all NWIS sites
+nwis <- listNWIS(aoi = well_aoi, dist = .3) 
+
+# Get NWIS Stream Gages
+ref_gages <- get_gagesII(id = unique(nwis$site_no)) %>%
```

Comment:
Confirmed that reference gage is the nearest gage on the Jemez River --- just curious.

Reply:
One question that this brought up, which we can discuss online and later incorporate into the workflow if needed, is whether reference stream gauges are truly the only ones we care about. I need to read the reference gauge paper but -- for example, would there ever be a gauge upstream from a reference gauge? Just putting this here

Reply:
Reference gages are "natural" gages mostly undisturbed. However gagesII was done a long time ago, and there are some things that weren't accounted for. For example, some of these gages have transbasin diversions. There could in theory by a gage upstream of a reference gage. Happy to chat more about this offline:)